### ISSUE TITLE
ListColumnData::ScanCount - internal list scan offset is out of range

### ISSUE THREAD

**Issue by pkese** [duckdb/duckdb#10212]
### What happens?

I've started testing DuckDb and have imported some 90 million rows of entities from https://dumps.wikimedia.org/wikidatawiki/entities/

This is what my table looks like:
```
select * from items where id > 'P' limit 10;
┌─────────┬─────────┬───────────────────┬──────────────────────┬──────────────────────┬───────────────────────────────────┬─────────────────────┬───────────┬────────────────────────────────────────────────────────────┬───────────────────┐
│   id    │  type   │       label       │     description      │       aliases        │            instance_of            │     subclass_of     │ facet_of  │                          part_of                           │    wiki_title     │
│ varchar │ varchar │      varchar      │       varchar        │      varchar[]       │             varchar[]             │      varchar[]      │ varchar[] │                         varchar[]                          │      varchar      │
├─────────┼─────────┼───────────────────┼──────────────────────┼──────────────────────┼───────────────────────────────────┼─────────────────────┼───────────┼────────────────────────────────────────────────────────────┼───────────────────┤
│ Q31     │ item    │ Belgium           │ country in western…  │ [Kingdom of Belgiu…  │ [Q3624078, Q43702, Q6256, Q2018…  │ []                  │ []        │ [Q215669, Q458, Q46, Q1065, Q13116, Q476033, Q7184, Q8932] │ Belgium           │
│ Q8      │ item    │ happiness         │ mental or emotiona…  │ [joy, happy]         │ [Q331769, Q60539479]              │ [Q16748867]         │ []        │ [Q77468620, Q13100823]                                     │ Happiness         │
│ Q23     │ item    │ George Washington │ president of the U…  │ [Father of the Uni…  │ [Q5]                              │ []                  │ []        │ [Q186539]                                                  │ George Washington │
│ Q24     │ item    │ Jack Bauer        │ character from the…  │                      │ [Q15632617, Q15773317, Q20085850] │ []                  │ []        │ []                                                         │ Jack Bauer        │
│ Q42     │ item    │ Douglas Adams     │ English author and…  │ [Douglas Noël Adam…  │ [Q5]                              │ []                  │ []        │ []                                                         │ Douglas Adams     │
│ Q1868   │ item    │ Paul Otlet        │ Belgian author, li…  │ [Paul Marie Ghisla…  │ [Q5]                              │ []                  │ []        │ []                                                         │ Paul Otlet        │
│ Q2013   │ item    │ Wikidata          │ free knowledge gra…  │ [wikidata.org, m.w…  │ [Q33120876, Q638153, Q36509592,…  │ []                  │ []        │ [Q54837, Q43984865]                                        │ Wikidata          │
│ Q45     │ item    │ Portugal          │ country in Southwe…  │ [Portuguese Republ…  │ [Q3624078, Q6256, Q20181813, Q1…  │ []                  │ []        │ [Q50807777, Q458, Q8932]                                   │ Portugal          │
│ Q51     │ item    │ Antarctica        │ polar continent in…  │ [Antarctic Selecti…  │ [Q5107, Q82794, Q312461]          │ []                  │ []        │ [Q205895, Q1555938, Q2]                                    │ Antarctica        │
│ Q58     │ item    │ penis             │ primary sexual org…  │                      │ [Q712378]                         │ [Q4620674, Q168552] │ []        │ [Q842083, Q181100]                                         │ Penis             │
├─────────┴─────────┴───────────────────┴──────────────────────┴──────────────────────┴───────────────────────────────────┴─────────────────────┴───────────┴────────────────────────────────────────────────────────────┴───────────────────┤
│ 10 rows                                                                                                                                                                                                                         10 columns │
└────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────┘
```

Ids are strings that can either contain a `'P'+<number>` or `'Q'+<number>`, so I was interested in what's in there:
```
D select count(*) from items where id > 'Q';
┌──────────────┐
│ count_star() │
│    int64     │
├──────────────┤
│     90170415 │
└──────────────┘
```

However this fails
```sql
D select * from items where id < 'Q' LIMIT 10;
Error: INTERNAL Error: ListColumnData::ScanCount - internal list scan offset is out of range
```

### To Reproduce

I can share the file, but it's rather large: 8 GB

### OS:

Ubuntu 23.10

### DuckDB Version:

v0.9.2 3c695d7ba9 (official binary from DuckDb.org)

### DuckDB Client:

DuckDb command line client (SQL)

### Full Name:

Peter Kese

### Affiliation:

/

### Have you tried this on the latest `main` branch?

I have tested with a main build

### Have you tried the steps to reproduce? Do they include all relevant data and configuration? Does the issue you report still appear there?

- [X] Yes, I have

**Comment by pkese** [comment]
I was interested in records starting with 'P'...

```sql
select * from items where id < 'Q'
```
^^ didn't work as shown above.

This doesn't work either:
```sql
D select * from items where id[0] = 'P' limit 10;
┌─────────┬─────────┬─────────┬─────────────┬───────────┬─────────────┬─────────────┬───────────┬───────────┬────────────┐
│   id    │  type   │  label  │ description │  aliases  │ instance_of │ subclass_of │ facet_of  │  part_of  │ wiki_title │
│ varchar │ varchar │ varchar │   varchar   │ varchar[] │  varchar[]  │  varchar[]  │ varchar[] │ varchar[] │  varchar   │
├────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────┤
│                                                         0 rows                                                         │
└────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────┘
```

This works:
```sql
D select * from items where starts_with(id, 'P') limit 10;
┌─────────┬──────────┬─────────────────────┬──────────────────────┬──────────────────────────────────────────────────────────────────────────────────┬────────────────────────────────────┬─────────────┬───────────┬───────────┬────────────┐
│   id    │   type   │        label        │     description      │                                     aliases                                      │            instance_of             │ subclass_of │ facet_of  │  part_of  │ wiki_title │
│ varchar │ varchar  │       varchar       │       varchar        │                                    varchar[]                                     │             varchar[]              │  varchar[]  │ varchar[] │ varchar[] │  varchar   │
├─────────┼──────────┼─────────────────────┼──────────────────────┼──────────────────────────────────────────────────────────────────────────────────┼────────────────────────────────────┼─────────────┼───────────┼───────────┼────────────┤
│ P461    │ property │ opposite of         │ item that is in so…  │ [inverse, opposite, is the opposite of, inverse of, contrasted to, contrast of…  │ [Q28100549]                        │ []          │ []        │ []        │            │
│ P467    │ property │ legislated by       │ indicates that an …  │ [enacted by, passed by]                                                          │ [Q22984026]                        │ []          │ []        │ []        │            │
│ P497    │ property │ CBDB ID             │ identifier for a p…  │                                                                                  │ [Q97584729, Q62589316]             │ []          │ []        │ []        │            │
│ P512    │ property │ academic degree     │ academic degree th…  │ [degree, diploma, academic diploma]                                              │ [Q18608871, Q22984450]             │ []          │ []        │ []        │            │
│ P517    │ property │ interaction         │ subset of the four…  │                                                                                  │ [Q22981316, Q119301849]            │ []          │ []        │ []        │            │
│ P533    │ property │ target              │ target of an attac…  │ [military target, target of attack, attack target]                               │ [Q22964785]                        │ []          │ []        │ []        │            │
│ P555    │ property │ doubles record      │ win/lose balance f…  │                                                                                  │ [Q21099935, Q27886949, Q28106456]  │ []          │ []        │ []        │            │
│ P557    │ property │ DiseasesDB          │ identifier sourced…  │ [Diseases Database]                                                              │ [Q19887775, Q18614948, Q110682839] │ []          │ []        │ []        │            │
│ P560    │ property │ direction           │ qualifier to indic…  │ [facing]                                                                         │ [Q15720608]                        │ []          │ []        │ []        │            │
│ P561    │ property │ NATO reporting name │ official reporting…  │ [ASCC reporting name, NATO code name]                                            │ [Q51089849, Q19847637]             │ []          │ []        │ []        │            │
├─────────┴──────────┴─────────────────────┴──────────────────────┴──────────────────────────────────────────────────────────────────────────────────┴────────────────────────────────────┴─────────────┴───────────┴───────────┴────────────┤
│ 10 rows                                                                                                                                                                                                                         10 columns │
└────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────┘

```

**Comment by szarnyasg** [comment]
@pkese please share the file! You can reach me via email at gabor@duckdblabs.com

**Comment by pkese** [comment]
@szarnyasg 
I have uploaded the file [here](http://srv.kese.net/wikidata/).

There are two files, the DuckDb database and original json files that I imported into the database as
```sql
CREATE TABLE items AS SELECT * FROM read_json_auto('raw/0000000.jsonl', records=true);
INSERT INTO items ... {remaining .jsonl files}
```

Sorry for my late response and thanks for your support.

**Comment by szarnyasg** [comment]
@pkese no problem, thanks for sharing the data set. I managed to reproduce it!

**Comment by Mytherin** [comment]
Thanks for the report! I've pushed a fix in #10268.
